utils/MigrateData.py

```diff
@@ -0,0 +1,802 @@
+        if not os.path.exists('KoalaDBBackups\\backup_' + file_number):
+            os.makedirs('KoalaDBBackups\\backup_' + file_number)
+        des = os.getcwd() + '\\KoalaDBBackups\\backup_' + file_number
+        regex = re.compile('.*Koala.db')
+        for root, dirs, files in os.walk(os.getcwd()):
+            for file in files:
+                if regex.match(file):
+                    src = os.path.join(os.getcwd(), file)
+                    shutil.copy(src, des)
+                    return True
+    except Exception as e:
+        print(e)
+        return False
+
+
+def get_largest_file_number():
+    """
+
+    :return:
+    """
+    src = os.getcwd() + '\\KoalaDBBackups'
+    if len(os.listdir(src)) == 0:
+        return 0
+    else:
+        values = [int(i[7:]) for i in os.listdir(src)]
+        values.sort()
+        return values[-1]
+
+
+def reset_db():
+    """
+    Deletes an errored Koala.db database in the cwd and replaces it with the most recently saved database
+    :return:
+    """
+    last_db = os.listdir(os.getcwd() + "KoalaDBBackups\\backup_" + str(get_largest_file_number()))[0]
+    os.remove(os.getcwd() + '\\Koala.db')
+    regex = re.compile('.*Koala.db')
+    for root, dirs, files in os.walk(os.getcwd()):
+        for file in files:
+            if regex.match(file):
+                os.remove(file)
+    shutil.copy(last_db, os.getcwd())
+
+
+class MigrateData:
+
+    def __init__(self, database_manager):
+        """
+        Initalised shit
+        :param database_manager: idk, jack help plz
+        """
+        self.database_manager = database_manager
+
+    def execute_update(self):
+        """
+        Sequentially applied the database update, if an error occurs then the entire database is rolled back.
+        :return:
+        """
+        if backup_data():
+            funcs = [self.remake_guilds, self.remake_guild_extensions, self.remake_guild_welcome_messages,
+                     self.remake_votes, self.remake_vote_sent, self.remake_vote_options, self.remake_vote_target_roles,
+                     self.remake_verified_emails, self.remake_not_verified_emails, self.remake_role_table,
+                     self.remake_to_re_verify, self.remake_twitch_alerts, self.remake_user_in_twitch_alert,
+                     self.remake_team_in_twitch_alert, self.remake_user_in_twitch_team, self.remake_text_filter,
+                     self.remake_text_filter_moderation, self.remake_text_filter_ignore_list,
+                     self.remake_guild_rfr_messages, self.remake_rfr_message_emoji_roles,
+                     self.remake_guild_rfr_required_roles, self.remake_guild_colour_change_permissions,
+                     self.remake_guild_invalid_custom_colour_roles, self.remake_guild_usage]
+            for func in funcs:
+                try:
+                    func()
+                except Exception as e:
+                    print(e)
+                    reset_db()
```

Comment:
This error could also be thrown again, or continued out of this function so that the rest of KoalaBot doesn't run with an invalid database.